### DISCUSSION TITLE
How can we add Labels and Annotations to Traits using KubeVela?

### DISCUSSION THREAD

**Discussion by MaharaJothi** [kubevela/kubevela#3919] [Q&A] [answered]
Hi

I want to add Labels and Annotations to my Traits (Ingress to be specific)

## Issue Description
I'm using Kong as Ingress controller in my K8s cluster.  To specify which ingress class to use and also to enable OIDC I need to add Annotations at Ingress file.

When using KubeVela, the labels and annotations that I add in my Trait Definition is not applied to the Ingress service created via the same definition.  

- Labels and Annotations added at the Component Definition is getting applied to its resources
- [Labels and Annotation](https://kubevela.io/docs/end-user/traits/annotations-and-labels) Trait provided by KubeVela is applied only to the Components not for any other types of resource

Is there a way or any support provided by KubeVela to achieve this?

Thanks in advance

**Comment by FogDong** [comment, marked as answer]
Hi, you can try to use `patchOutputs` in your trait definition to patch annotation to the tratis. Notice that you have to upgrade your KubeVela to 1.4(unreleased yet, or you can use master branch) to use it. You can also refer to this doc: https://kubevela.io/docs/next/platform-engineers/traits/patch-trait#patch-to-traits

**Reply by MaharaJothi** [reply]
Thank you @FogDong .  With the latest code from master branch, I was able to achieve this.
